api/routes/users/post.js

```diff
@@ -0,0 +1,45 @@
+const bcrypt = require('bcryptjs');
+const defaultDB = require('../../db')();
+const loggedIn = require('../../auth/middleware').loggedIn;
+
+const ifUserIsNew = (email, res, db) =>
+  db('users')
+    .where({ email })
+    .first()
+    .then(user => {
+      if (user) {
+        res
+          .status(400)
+          .send({ error: 'add-user-email-exists' })
+          .end();
+        return Promise.reject();
+      }
+      return Promise.resolve();
+    });
+
+const insert = (email, password, db) => {
+  const hashed = bcrypt.hashSync(password);
+  return db('users').insert({ email, password: hashed });
+};
+
+module.exports = (app, db = defaultDB) => {
+  app.post('/user', loggedIn, (req, res) => {
+    if (req.body.email && req.body.password) {
+      ifUserIsNew(req.body.email, res, db)
```

Comment:
malformed emails might be another good one to explicitly handle, but we can fold those in l8er